indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/Asn1Elements.kt

```diff
@@ -233,6 +304,50 @@ sealed class Asn1Element(
         }
 
         override fun hashCode(): Int = encodedTag.contentHashCode()
+
+        /**
+         * creates a new Tag from this object, overriding the class. Useful for implicitTagging (see [Asn1Structure.withImplicitTag])
+         */
+        infix fun withClass(tagClass: TagClass) =
+            Tag(this.tagValue, tagClass = tagClass, constructed = this.isConstructed)
+
+        /**
+         * creates a new Tag from this object, negating the passed property. Useful for implicitTagging (see [Asn1Structure.withImplicitTag]).
+         * This is a NOOP for tag that don't have this bit set.
+         */
+        infix fun without(negated: TagProperty): Tag = when (negated) {
```

Comment:
did i miss it, or is `with` missing?

Reply:
why would you invite the creation of illegal ASN.1 structures that don't parse?

Reply:
ah, right

should `Tagged(42)` maybe be `Constructed(tag = 42)`?

Reply:
please provide a more comprehensive example, as I have a hard time wrapping my head around this suggestion (and maybe check [RFC 5280](https://datatracker.ietf.org/doc/html/rfc5280) and how your proposal aligns with the syntax and lingo there)

Reply:
done and ready for review

Reply:
should there be an `ExplicitlyTagged(tag: Asn1.Tag)` overload that checks if the tag is explicit?

Reply:
It would probably make more sense, to add `Structure(tag:Tag)`, letting you specify whatever you want, that returns a custom structure

Reply:
`Sequence{…} withImplicitTag` does that and does not get in your way with tag-altering checks

Reply:
thing is, if you override anything but the numeric tag value, it is not an explicit tag any more, so it really does not make sense, because you'll end up with a custom structure

Reply:
what if i want to take the (explicit) tag of some structure and create a new structure with the same (explicit) tag?

Reply:
I'm thinking `fun ExplicitlyTagged(tag: Asn1.Tag, foo: ()->whatever) { require(tag.isExplicitTag); return ExplicitlyTagged(tag.tagValue, foo); }`

Reply:
> what if i want to take the (explicit) tag of some structure and create a new structure with the same (explicit) tag?

`Asn1.ExplicitTag(oldSctrucuter.tag.tagValue) {…}`

Again, the only variable for this DSL invocation is the numeric tag value, all other bits are fixed, so it makes no sense to allow passing other properties, just to cause confusion and invite user errors